src/Symfony/Bundle/FrameworkBundle/Test/BrowserKitAssertionsTrait.php

```diff
@@ -124,6 +126,19 @@ public static function assertRouteSame($expectedRoute, array $parameters = [], s
         self::assertThat(self::getRequest(), $constraint, $message);
     }
 
+    public static function assertThatForResponse(Constraint $constraint, string $message = ''): void
+    {
+        try {
+            self::assertThat(self::getResponse(), $constraint, $message);
+        } catch (ExpectationFailedException $exception) {
+            if ($serverExceptionMessage = self::getResponse()->headers->get('X-Debug-Exception')) {
+                $exception->__construct($exception->getMessage(), $exception->getComparisonFailure(), new \Exception($serverExceptionMessage));
```

Comment:
so, let's improve this a bit with an ErrorException now, and let ErrorHandler report the file and the line too, to populate ErrorException's constructor with file+line info.

Reply:
Updated.